devtools/ci/gh-actions/conda-envs/docs.yml

```diff
@@ -13,7 +13,7 @@ dependencies:
 - swig
 - doxygen 1.8.14
 - pip:
-  - sphinx==2.3.1
-  - sphinxcontrib-bibtex<2.0.0
+  - sphinx==4.0.2
+  - sphinxcontrib-bibtex
   - sphinxcontrib-lunrsearch
   - sphinxcontrib-autodoc_doxygen
```

Comment:
CI presently installs both of the affected packages via PIP, so the new tags have already taken effect. This PR is successfully building the docs, unlike `master` which was broken by the new releases of `sphinxcontrib-autodoc_doxygen` and `sphinxcontrib-lunrsearch`. The status failure is in a subsequent check that looks for broken links, not in building the docs.

I think this PR needs to be merged ASAP and is not blocked by upstream releases.